### PR TITLE
test: coverage push with shared atomicfile package

### PR DESCRIPTION
**Pull request by luckyPipewrench**
## Summary
- Extract shared `internal/atomicfile` package from 3 duplicate atomic write implementations (signing, integrity, rules), enabling DI-based testing of write/chmod/close error paths
- Add ~150 test cases across 20+ packages targeting functions below 95% coverage
- Remove unused `perm` parameter from `rules.atomicWriteFile` (unparam lint fix)

Key per-package improvements:
| Package | Before | After |
|---------|--------|-------|
| enterprise | 92.9% | 98.0% |
| addressprotect | 93.8% | 96.5% |
| projectscan | 95.8% | 98.2% |
| license | 91.0% | 94.0% |
| sentry | 97.5% | 99.2% |
| mcp/tools | 98.2% | 99.7% |
| metrics | 98.5% | 99.0% |

Overall: 88.3% → 88.7%. Remaining floor is CLI (needs package split) and sandbox (needs kernel primitives).

## Summary by CodeRabbit

* **Refactor**
  * Centralized atomic file-write behavior for safer, consistent on-disk updates used by manifests, locks, and signing — improves reliability of persisted files and permissions.

* **Tests**
  * Vastly expanded test coverage across config validation, licensing/audit, CLI, proxy/listeners, sandboxing, signing/keystore, integrity, and many utilities to catch edge cases and regressions earlier.

### CODE REVIEW

File: internal/scanner/coverage_test.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+	}
+
+	// Very small window: interval floors at 1s.
+	fb2 := NewFragmentBuffer(65536, 1000, 1)
+	defer fb2.Close()
+	fb2.Append("s2", []byte("data"))
+	if fb2.TotalBufferBytes() == 0 {
+		t.Error("expected non-zero bytes")
+	}
+}
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply internal/scanner/coverage_test.go:152]: The cleanup interval is stored in a private struct field and isn't accessible from tests without exporting it. These tests exercise the clamping code path (the coverage goal), and the clamped values are validated by the constructor's internal logic. Adding external assertions would require exporting internal state solely for testing, which we avoid in this codebase.

File: internal/signing/coverage_boost_test.go [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+			info, err := os.Stat(path)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if info.Mode().Perm() != tt.perm {
+				t.Errorf("permissions = %04o, want %04o", info.Mode().Perm(), tt.perm)
+			}
+		})
+	}
+}
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply internal/signing/coverage_boost_test.go:104]: The 0o644 case is intentional — it proves `atomicfile.Write` applies the caller's requested permission rather than hardcoding 0o600. Without testing a non-default permission, we can't distinguish "function respects the perm parameter" from "function always writes 0o600." Production callers all use 0o600 per project policy, but the shared utility must be correct for any valid permission.
